pandas/core/arrays/datetimes.py

```diff
@@ -586,7 +586,8 @@ def astype(self, dtype, copy=True):
             # GH#18951: datetime64_ns dtype but not equal means different tz
             new_tz = getattr(dtype, "tz", None)
             if getattr(self.dtype, "tz", None) is None:
-                return self.tz_localize(new_tz)
+                # GH 33401: Match the behavior of Series.astype
```

Comment:
is it obvious that Series is the better behavior?  it might be that the only unambiguous thing to do is raise and require the user to be explicit

Reply:
That's fair. The implicit UTC localization isn't my favorite behavior.

With that in mind, should we deprecate the `Series.astype(tz_dtype)` UTC localization behavior instead?